code/mob/dead/ai-camera.dm

```diff
@@ -437,6 +445,97 @@
 		if(mainframe)
 			mainframe.ai_station_announcement()
 
+	verb/ai_open_map()
+		set category = "AI Commands"
+		set name = "Open AI map"
+		set desc = "Open a map of the station, double click on the map to teleport"
+
+		//default to cogmap if somehow the map settings go missing
+		//currently only Ozymandias and Icarus are unsupported
+		var/map_name = "cogmap"
+		if (map_settings)
+			var/position = findtext(map_settings.goonhub_map, "/maps/")
+			if (position > 0)
+				map_name = copytext(map_settings.goonhub_map, position + length("/maps/"))
+		//html dumped into source files fuck yeah
+		var/HTML = {"
+			<!DOCTYPE html>
```

Comment:
yeah no

please make this into a tgui interface - it even have a special byond ui component to help with this

Reply:
Okay, cool. I am entirely on board with using modern UI frameworks. I'll start porting it now.

Reply:
Just to clarify, do you mean make a tgui interface that that renders the website map, or one that uses a byond map component to render the map itself?

Reply:
Either - but seeing as the former is what you did here, then the former.
Thinking on this again, you probably don't need the byondui component to catch the clicks or anything.